app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -68,11 +68,22 @@
 
     private boolean batterySaver;
 
-    private boolean pokeFlyRunning = false;
     private int trainerLevel;
 
+    private Button launchButton;
+
     private final Point arcInit = new Point();
     private int arcRadius;
+    private final BroadcastReceiver pokeflyStateChanged = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            if (Pokefly.ACTION_START.equals(intent.getAction())) {
+                updateLaunchButtonText(true);
+            } else if (Pokefly.ACTION_STOP.equals(intent.getAction())) {
+                updateLaunchButtonText(false);
+            }
```

Comment:
Can't we just handle it with one Intent? And just pass true/false based on Pokefly.isRunning()?